### PR TITLE
PLT-3960 Don't show (message deleted) placeholder to the user that deleted the message

### PR DESCRIPTION
**Pull request by khawerrind**
**Summary**

When someone deletes the post `(message deleted)` was showing even for the user who deleted the post. Now the owner of the post will not see this message and the post will `disappear` for him. While other users will still see `(message deleted)` message
**Ticket Link**

https://mattermost.atlassian.net/browse/PLT-3960

### CODE REVIEW

**Comment by yangchen1** [comment]
Hey @khawerrind, this overall looks good! Thanks for submitting another PR. I think there may have been some confusion around the ticket - it seems like you implemented it so the (Message Deleted) doesn't show for the _poster_ of the deleted message. I think what we actually want is for (Message Deleted) not to show for the _deleter_ of the deleted message. Most of the time, the deleter and the poster is same person - but when an admin deletes someone else's message, we want the (deleted message) to still show for the _poster_, but not for the admin. Does that make sense? :)

**Comment by khawerrind** [comment]
@yangchen1 fixed.

**Comment by yangchen1** [comment]
Looks good to me. 👍 Great work! Tested deleting replies, DM's, and admin deleting other people's messages.

**Comment by esethna** [comment]
Hi @khawerrind thanks for the PR! After reviewing I found an issue when Team Admins attempt to delete messages posted by others or their own messages. See GIF below. What happens is that the "(message deleted)" shows breifly until a new action is completed.

delete messages

____

I didn't find any other issues. I tested deleting from the right hand-side and center channel, compact view, messages with emoji reations, public channels, private channels, DMs, and all permission levels (system admin, team admin, member).

**Comment by khawerrind** [comment]
@esethna fixed the bug. Kindly check again. Thanks!

**Comment by yangchen1** [comment]
+1 from me :)

File: webapp/components/delete_post_modal.jsx [hunk 1]
```diff
@@ -54,8 +54,8 @@ export default class DeletePostModal extends React.Component {
             this.state.post.channel_id,
             this.state.post.id,
             () => {
-                PostStore.deletePost(this.state.post);
-
+                PostStore.removePost(this.state.post);
```
Comment (crspeller) [inline webapp/components/delete_post_modal.jsx]: This should really be an action. You can see examples here: https://github.com/mattermost/platform/blob/master/webapp/actions/post_actions.jsx#L88

You can find more info on our architecture (flux) here: https://www.youtube.com/watch?v=8dB2_29TtNo

We would really appreciate you turning this into an action.

**Comment by khawerrind** [comment]
@crspeller added it to global action. Please check. Thanks!

**Comment by esethna** [comment]
Hi @khawerrind, thanks for the tweaks!

Just letting you know there may be a delay in reviewing and merging this PR as we are locking down feature and improvement contributions for the v3.6 release, per our [release process](https://docs.mattermost.com/process/release-process.html). This PR can be merged into `master` after we branch `release-3.6` on January 4th.

**Comment by IamAbhiKaushik** [comment]
Is there any way Admin can decide to enable disable this feature for ALL Users? 

for example, if set to false, No one will see the `message deleted` prompt when a message was deleted.
I was making a bot that send alerts on mattermost, and we need to delete the stale conversations, is there any way of not showing all these deleted message alerts??

I am looking for something apart from page reload.

**Comment by bearcolonel** [comment]
> Is there any way Admin can decide to enable disable this feature for ALL Users?
> 
> for example, if set to false, No one will see the `message deleted` prompt when a message was deleted.
> I was making a bot that send alerts on mattermost, and we need to delete the stale conversations, is there any way of not showing all these deleted message alerts??
> 
> I am looking for something apart from page reload.

If you find a method to solve your problem, plz, answer)

**Pull request merged**